source/agora/common/BlockStorage.d

```diff
@@ -120,18 +230,27 @@ public class BlockStorage
         Params:
             findex = the index of the file.
 
+        Returns:
+            Returns true if success, otherwise returns false.
+
     ***************************************************************************/
 
     private bool map (size_t findex) @trusted nothrow
     {
         try {
-            if ((this.file !is null) && (findex == this.file_index))
-                return true;
-
             if (this.file !is null)
+            {
+                if (findex == this.file_index)
+                    return true;
+
+                if (this.is_saving)
+                    assert(this.writeChecksum());
```

Comment:
Like Mathias mentioned, this assert will disappear in `-release` mode. It needs to be replaced with:

```
if (this.is_saving && !this.writeChecksum())
    assert(0);
```

`assert(0);` is special. It is similar to `abort();`.

Reply:
Applied the suggestion